### PR TITLE
Merge definitions from .services.yml properly.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

**Comment by dawehner** [comment]
Let's ensure we have test coverage for this rather common usecase

**Comment by drupol** [comment]
Writing tests...

File: modules/providers/service_container_symfony/lib/Drupal/service_container_symfony/Tests/ServiceContainerSymfonyTest.php [hunk 1]
```diff
@@ -26,17 +26,28 @@ public static function getInfo() {
    * {@inheritdoc}
    */
   protected function setUp() {
-    $modules[] = 'test_service_container_symfony';
+    $modules[] = 'service_container_symfony_test';
+    $modules[] = 'service_container_symfony_subtest';
     parent::setUp($modules);
+
+    print_r($this->container->getDefinitions());
```
Comment (fabianx-ai) [inline modules/providers/service_container_symfony/lib/Drupal/service_container_symfony/Tests/ServiceContainerSymfonyTest.php]: debug code.

File: modules/providers/service_container_symfony/tests/modules/service_container_symfony_subtest/service_container_symfony_subtest.services.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+services:
```
Comment (fabianx-ai) [inline modules/providers/service_container_symfony/tests/modules/service_container_symfony_subtest/service_container_symfony_subtest.services.yml:1]: Don't we want to test parameter merging, too?

**Comment by fabianx-ai** [comment]
Tests need work, the patch looks great! :)

**Comment by drupol** [comment]
Busy writing the parameters test already.

**Comment by fabianx-ai** [comment]
Two tests passed, merging in.

**Pull request merged**